service/service.go

```diff
@@ -280,7 +282,25 @@ func (s *Service) loop(ctx context.Context, roundToResume *round) error {
 			if err := s.openRound.teardown(false); err != nil {
 				return fmt.Errorf("tearing down open round: %w", err)
 			}
-			return nil
+			_ = eg.Wait()
```

Comment:
```suggestion
			eg.Wait()
```

Reply:
Then the linter configuration is incorrect, `golangci-lint` does not complain if `_ =` is removed.

`eg.Wait()` will never return an error, because no routine started with it returns anything but `nil` 😉. Does your linter also complain that 

```go
fmt.Println("Hello, World!")
```

should be

```go
_, _ = fmt.Println("Hello, World!")
```

Should the return values of print, close, write, read, etc. always EXPLICITLY be ignored?

Reply:
> eg.Wait() will never return an error, because no routine started with it returns anything but nil :wink:.

Yes, that's why it's fine to ignore the error with `_ = `. I can't do anything with `gosec` being inconsistent. Perhaps it doesn't detect an unhandled error if there is more than 1 return value?

Reply:
I think maybe using an `errgroup` here is why the workaround is needed in the first place. Maybe `sync.Waitgroup` would be a better fit?

`errgroup` is basically a wrapper around `sync.Waitgroup` to extend it with error propagation, and Context cancelation. If that is not needed, maybe the Waitgroup would work instead?

```golang
var wg sync.Waitgroup
wg.Add(1)
go func() {
    defer wg.Done()
    // some task
}

wg.Wait() // doesn't return an err
```

Reply:
I avoid Waitgroup because it requires manually calling `.Add()` and `.Done()` which is error-prone.

Reply:
I tend to disagree. There isn't much difference between

```golang
eg.Go(func() error {
	// task
	return nil
})
eg.Wait()
```

and

```golang
wg.Add(1)
go func() {
	defer wg.Done()
	// task
}()
wg.Wait()
```

it also has the advantage of being more performant because of the overhead that `errgroup` has for the error handling that we aren't using anyway and built in cancelation. Its also easier to scope variables if needed, since they can be passed to the goroutine via parameters, which for the `errgroup` takes more effort.

Reply:
IMHO, there is a huge difference. In the second snippet, it's easy to forget `wg.Add()` or `wg.Done()` and create buggy code. This is not possible with the errgroup.

> Its also easier to scope variables if needed.

I don't know what you mean.